extensions/indexes/lucene/src/org/exist/indexing/lucene/LuceneIndexWorker.java

```diff
@@ -300,7 +300,7 @@ public MatchListener getMatchListener(DBBroker broker, NodeProxy proxy) {
         boolean needToFilter = false;
         Match nextMatch = proxy.getMatches();
         while (nextMatch != null) {
-            if (nextMatch.getIndexId() == LuceneIndex.ID) {
```

Comment:
Need to think about this; probably we can use object reference comparison here which is faster

Reply:
Object reference comparison is not really a safe idea here, as these are strings and strings are immutable copies.

Reply:
for sure you are right, strictly speaking. But actually here a string reference check would be ok enough :-)